### PR TITLE
Convert admin api docs to markdown

### PR DESCRIPTION
**Pull request by anoadragon453**
So that they render nicely in mdbook (see #10086), and so that we no longer have a mix of structured text languages in our documentation (excluding files outside of `docs/`).

The largest change in here is definitely the `user_admin_api` docs.

### CODE REVIEW

File: docs/admin_api/purge_history_api.md [hunk 1]
```diff
@@ -13,65 +12,67 @@ delete the last message in a room.
...
 
-``POST /_synapse/admin/v1/purge_history/<room_id>[/<event_id>]``
+```
+POST /_synapse/admin/v1/purge_history/<room_id>[/<event_id>]
+```
 
-To use it, you will need to authenticate by providing an ``access_token`` for a
-server admin: see `README.rst <README.rst>`_.
+To use it, you will need to authenticate by providing an `access_token` for a
+server admin: see [Admin API](../../usage/administration/admin_api/index.html).
```
Comment (richvdh) [inline docs/admin_api/purge_history_api.md:20]: is this link expected to work? It's a 404 right now.

Comment (richvdh) [reply docs/admin_api/purge_history_api.md:20]: likewise a bunch of similar links below.

**Review by richvdh — APPROVED** [review]
I haven't reviewed every line of `user_admin_api`, but it looks good other than the below.

**Comment by richvdh** [comment]
It's going to conflict with #10044 I'm afraid.

File: docs/admin_api/purge_history_api.md (hunk 1 above)
Comment (anoadragon453) [reply docs/admin_api/purge_history_api.md]: Ah, there's two things here:

* That link only works in the context of https://github.com/matrix-org/synapse/pull/10086 which this commit was pulled out of.
* We shouldn't link to `...index.html` as that won't work when viewing the docs on GitHub. It should have been `../../usage/administration/admin_api/README.md`. I'll keep this in mind when updating https://github.com/matrix-org/synapse/pull/10086 once this has merged.

**Comment by anoadragon453** [comment]
> It's going to conflict with #10044 I'm afraid.

Actually, this ended up not doing so as only a markdown file was changed :slightly_smiling_face:

**Pull request merged**